time/src/test/java/net/jqwik/time/api/PeriodTests.java

```diff
@@ -126,12 +144,124 @@ void daysBetweenSame(@ForAll int start, @ForAll Random random) {
 
 	}
 
+	@Group
+	class Shrinking {
+
+		@Property
+		void defaultShrinking(@ForAll Random random) {
+			PeriodArbitrary periods = Dates.periods();
+			Period value = falsifyThenShrink(periods, random);
+			assertThat(value).isEqualTo(Period.of(0, 0, 0));
+		}
+
+		@Disabled
+		@Property
+		void shrinksToSmallestFailingValue(@ForAll Random random) {
+			PeriodArbitrary periods = Dates.periods();
+			TestingFalsifier<Period> falsifier = period -> comparePeriodsWithMax30DaysAnd11Months(period, Period.of(200, 3, 5)) <= 0;
+			Period value = falsifyThenShrink(periods, random, falsifier);
+			assertThat(value).isEqualTo(Period.of(200, 3, 5));
+		}
```

Comment:
Again it doesn't shrink perfectly. It shrinks to `Period.of(201, 0, 0)`

Reply:
Using only days (and calculating it from days, months and days) would improve that.

Reply:
Yeah, but the problem here is, that you can't calculate it from days I guess:

- years/months/days can be negative and positive (if the user set it), so there are more solutions for one value
- months have different length and you don't know which month it is

Reply:
I have an idea how you can calculate it from one number, but this number has to be very large. I'll try it

Reply:
If it doesn't work we have to think of something else.

Reply:
After experimenting a bit with the Period API and how it's supposed to work, I guess your first approach was right and I'll have to tweak shrinking a bit.